### ISSUE TITLE
Responsive theme

### ISSUE THREAD

**Issue by madzgo** [SimpleMachines/SMF#272]
Hey everyone, this is my first post on github :)

Anyway, I have a suggestion  -  SMFs default theme to be responsive.

We all know that Responsive web design is the best way to handle support for various screen sizes.

By the way, I could help with front-end development. 
Can someone tell me - is there a specified design for SMF 2.1? Can I make a suggestion?

**Comment by colinschoen** [comment]
Welcome to github.

What do you mean is there a specified design for SMF 2.1? Download the current build and have a look. If you think something needs changing make a comment and send a PR.

**Comment by joshuaadickerson** [comment]
@madzgo, there are no mockups or general ideas to the theme. The way it
generally goes around here is: come up with something good and do the work
to make it happen and it has a shot at becoming the theme. I wish you luck,
UI design is the hardest thing for us.

**Comment by emanuele45** [comment]
> there are no mockups or general ideas to the theme. 

And this is a quite big issue TBH.
We don't have public plans/ideas/mockups not only for the theme, but for (almost) anything. And without any public plan it's a bit difficult for people that would like to help to find something to do (if madzgo creates a theme how could he be sure that it will be accepted and it will not result in quite a lot of wasted time?)

madzgo, the last idea I think it was to have two separated themes: one for "desktops" and one for "mobile devices". That way it would be easier to control the informations outputted and optimise the markup for mobile devices. I'm not a mobile-guru (neither a theme/accessibility expert, that was Ant), so I don't know what's the best option here...up to you guys.

**Comment by madzgo** [comment]
What I could do is make a home page design ( that would be boardindex and index templates )  - just in photoshop.

Then you guys can take a look and decide if its good for SMF.

If it passes, I could make a few more PSDs for most important page templates (primarily display and message index).

**Comment by matthew-kerle** [comment]
Go for it madzgo, obviously we can't guarantee anything, but I'd definitely love to see your ideas for the theme.

**Comment by moskis** [comment]
[single topic](https://f.cloud.github.com/assets/427979/54179/4c0f4cac-5a86-11e2-972c-1a89c3817d67.png)
[topic list](https://f.cloud.github.com/assets/427979/54180/4c7a3b84-5a86-11e2-8f55-6b4dbc861f9f.png)

Here's a couple of mockups i was doing in photoshop a couple of weeks ago. There's nothing responsive here since it's just photoshop but i wanted to try to make a clean and simple design that goes along with semantic HTML and good CSS classes so that it can be easy to customize and a good foundation for other themes/skins.

I did some HTML for it too and have been considering if doing a theme on my own out of it, but since that is a lot of work i'm not sure of it yet.

In any case i wanted to share them here in case they serve as any inspiration to you.

**Comment by emanuele45** [comment]
Looks a bit like github.

Just to be more clear: the default theme is the one already here, it needs to be finished (missing templates, etc.), but it should not changed much (unless there are good reasons to).

There is space for a mobile theme that was in the plans and at the moment we don't have (so the plan B is to just drop it from 2.1 features list: better ship something without a piece that not to ship anything).

**Comment by joshuaadickerson** [comment]
I think the next theme we create should make use of a grid system like in Bootstrap (which I think @moskis' theme looks like).

**Comment by madzgo** [comment]
@emanuele45 , does that mean that a full redesign is not an option?

On 9 January 2013 20:12, Joshua Dickerson notifications@github.com wrote:

> I think the next theme we create should make use of a grid system like in
> Bootstrap (which I think @moskis https://github.com/moskis' theme looks
> like).
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/SimpleMachines/SMF2.1/issues/272#issuecomment-12061380.

## 

Sreten Madžgalj\* *
Developer
_Bild Studio_ d.o.o.
*
*
mobile +382/ 67 9 24 26 32
office +382/ 67 911 3 711
sreten.madzgalj@bild-studio.net
www.bild-studio.com

http://www.bild-studio.com/
_Bild Studio_ - We bild your business
See and try because we are different than others.

**Comment by joshuaadickerson** [comment]
Not for 2.1, but it is still open for 2.2 or any other version after that.

**Comment by emanuele45** [comment]
The current is already a (almost) full redesign of Curve (and it barely fits the original plans), I don't think we can afford a double redesign in a minor release...

**Comment by moskis** [comment]
My mockup is inspired by Google new(ish) interface as well as Github and Bootstrap is you guys pointed out. :)

Regarding the theme of future versions, in my opinion the most important part of a theme is the HTML. Good semantic HTML (maybe even HTML5), and good IDs and Classes that make forum design easier to customize through CSS would be great. This can already be done, but there is room for improvement.

In other words, i don't see many themes that come with custom template files, so the more elements you can edit with CSS alone, the better.

**Comment by XinYenFon** [comment]
Just for info Curve II currently using responsive in its some parts like

@ media screen and (max-width: 1220px) {
    .table_list td.stats {
        display: none;
    }
    .table_list td.lastpost p {
        border-left: 1px solid #ccc;
        padding: 1px 5px 2px 10px;
        margin-top: 1px;
    }

## }

Edit: Lol only in that part 
Beside that yes, SMF 2.2 has to aim HTML5

**Comment by matthew-kerle** [comment]
There is a chance that Curve II will utilize HTML5. I still have to look into how time consuming it will be to convert it all. If it's going to take too long, we will keep it XHTML and push HTML5 to a future version of SMF 2.x.

**Comment by Arantor** [comment]
A full proper conversion of Curve II to HTML5 is not really viable at this stage; we'd like a release out by the end of the year if possible, and a full workover with things like regression testing is not really viable in the time frame - 8 months ago when this was raised, sure.

There is of course no doubt that 3.0 should have fully HTML5 theme though I'm wary towards Responsive, it still means you have to process and output content you're not necessarily displaying which, for a forum, is a sizeable amount. Responsive in a blog is great, but in a forum there is so much more information that may be relevant that needs to fit in the space available.

**Comment by Xarcell** [comment]
I think consider all the stuff we can't do because of backwards compatibility, perhaps this idea is not viable at this stage. I'm requesting a closing of this issue(feature enhancement).

**Comment by Arantor** [comment]
Except that as it turns out, a full workover of the theme is required anyway because of the number of issues we've found...

**Comment by Xarcell** [comment]
OMG...

This is "almost" confusing. What kind of changes can we expect?

**Comment by Arantor** [comment]
Lots and lots of bug fixes. There's a distinct lack of consistency in styling across the theme, for example.
